### DISCUSSION TITLE
how to download file with cookies

### DISCUSSION THREAD

**Discussion by heyouquan** [hgouveia/node-downloader-helper#95] [General]
Environment：Electron
"node-downloader-helper": "^2.1.6"

Now, throw a error when download a file without cookies, throw 401 error.

**Comment by hgouveia** [comment]
> Environment：Electron "node-downloader-helper": "^2.1.6"
> 
> Now, throw a error when download a file without cookies, throw 401 error.

if you have the cookies stored you just need to specify it on the headers options

**Comment by heyouquan** [comment]
> 

The cookies are set to HttpOnly, can not read.how to specify it on the headers options?

**Reply by hgouveia** [reply]
you will need to add the header in the options like any other, for example

```js
const dl = new DownloaderHelper(url, dest, { header: { 'Cookie' : myCookies }} );
```
